### PR TITLE
add tests for Data Submission Portal

### PR DESCRIPTION
**Pull request by devincowan**
Initial tests for the [Data Submission Portal](https://github.com/cznethub/dspfront)

I went a little rogue with the Page Object Model because these tests will largely be replicated on other pages (for the Zenodo, Earthchem, etc repositories). So rather than re-defining all of the page elements for those pages, I tried to make the selectors a bit more adaptable. Most of the elements are selected on data-id because the Vue generates the forms dynamically from a [json schema](https://github.com/eclipsesource/jsonforms).

### CODE REVIEW

**Comment by devincowan** [comment]
@hydrocheck here are 14 initial tests for the Data Submission Portal
[test_A_000008](https://github.com/CUAHSI/QA-AutomationEngine/pull/61/files#diff-6815d7f4054de0a764a7b77f0868c67ae1e93877bf96368cd1b0ca86446a50e6R196) fails purposefully right now because there is work pending to fix a bug in DSP.
I haven't been able to get a QA workstation so have only run these locally.
Thus some of the timing might not be robust to run on the grid hub -- I will be happy to update that once we get the QA stations running!
I went a little rogue with the Page Object Model because these tests will largely be replicated on other pages (for the Zenodo, Earthchem, etc repositories). So rather than re-defining all of the page elements for those pages, I tried to make the selectors a bit more adaptable. Most of the elements are selected on `data-id` because the Vue generates the forms dynamically from a [json schema](https://github.com/eclipsesource/jsonforms).

**Comment by devincowan** [comment]
@ndebuhr sorry to keep pushing modifications here. If you're going to review, let me know and I will stop for a while

File: dsp/dsp.py [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+    def fill_ids_submit_and_check(self, sort_text, section, nth, dict):
+        """Fill additional fields of HS submit page based on 'data-id'
+        Then submit the form, search in 'My Submissions',
+        and check that all of the fields match what was entered
+        """
+        success_filling = SubmitHydroshare.fill_inputs_by_data_ids(self.driver, dict, section, nth)
+        self.assertTrue(success_filling)
+        SubmitHydroshare.finish_submission(self.driver)
+
+        # The page isn't sorted upon load
```
Comment (sblack-usu) [inline dsp/dsp.py:137]: Is this a bug that should be fixed instead of accounting for it in our automated tests?

File: dsp/dsp.py [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+                SubmitHydroshare.fill_text_by_page_property(self.driver, text_elem, auto_text)
+            self.assertTrue(SubmitHydroshare.is_finishable(self.driver))
+
+    def test_A_000006(self):
+        """Confirm that CREATOR is populated from HS profile"""
+        auto_text = time.strftime("%d_%b_%Y_%H-%M-%S", time.gmtime())
+        self.login_and_autofill_hs_required(auto_text)
+        SubmitHydroshare.finish_submission(self.driver)
+
+        # The page isn't sorted upon load
```
Comment (sblack-usu) [inline dsp/dsp.py:204]: accounting for that bug again.

File: dsp/dsp.py (hunk 1 above)
Comment (devincowan) [reply dsp/dsp.py:137]: yep I have an open ticket for it here:
https://github.com/cznethub/dspfront/issues/53

File: hydroshare/hs_macros.py [hunk 3]
```diff
@@ -1052,7 +1052,7 @@ class Resource(Hydroshare):
     abstract = SiteElement(By.ID, "id_abstract")
     abstract_save = SiteElement(By.CSS_SELECTOR, "#div_id_abstract button")
     public_resource_notice = SiteElement(By.ID, "missing-metadata-or-file")
-    subject_keywords = SiteElement(By.ID, "txt-keyword")
+    subject_keyword_input = SiteElement(By.ID, "txt-keyword")
```
Comment (sblack-usu) [inline hydroshare/hs_macros.py:1055]: Why was this changed?

Comment (devincowan) [reply hydroshare/hs_macros.py:1055]: I wanted to make it more obvious that this targets the input element (which is deeply nested within the Vuetify element) vs the entire subject keywords form element. But you're right this rename isn't necessary, this was important before I removed some of the other `subject keywords...` elements during refactoring

**Review by sblack-usu — APPROVED** [review]
Looks great.  I like the `nth` solution for the generated forms.  I also think we should fix the submissions page bug instead of accounting for it in our tests.

**Comment by devincowan** [comment]
thanks @sblack-usu, I just added a test for the sorting. It currently fails. This should ensure that we don't forget about this issue when we run tests--good call on your part thanks!
